Libraries/LibWeb/IndexedDB/Internal/Algorithms.cpp

```diff
@@ -790,4 +781,648 @@ void commit_a_transaction(JS::Realm& realm, GC::Ref<IDBTransaction> transaction)
+    };
+    store->store_a_record(record);
+
+    // 5. For each index which references store:
+    for (auto const& [name, index] : store->index_set()) {
+        // 1. Let index key be the result of extracting a key from a value using a key path with value, index’s key path, and index’s multiEntry flag.
+        auto index_key = TRY(extract_a_key_from_a_value_using_a_key_path(realm, value, index->key_path(), index->multi_entry()));
+
+        // 2. If index key is an exception, or invalid, or failure, take no further actions for index, and continue these steps for the next index.
+        if (index_key.is_error())
+            continue;
+
+        auto index_key_value = index_key.value();
+        auto index_multi_entry = index->multi_entry();
+        auto index_key_is_array = index_key_value->type() == Key::KeyType::Array;
+        auto index_is_unique = index->unique();
+
+        // 3. If index’s multiEntry flag is false, or if index key is not an array key,
+        //    and if index already contains a record with key equal to index key,
+        //    and index’s unique flag is true,
+        //    then this operation failed with a "ConstraintError" DOMException.
+        //    Abort this algorithm without taking any further steps.
+        if ((!index_multi_entry || !index_key_is_array) && index_is_unique && index->has_record_with_key(index_key_value))
+            return WebIDL::ConstraintError::create(realm, "Record already exists in index"_string);
+
+        // 4. If index’s multiEntry flag is true and index key is an array key,
+        //    and if index already contains a record with key equal to any of the subkeys of index key,
+        //    and index’s unique flag is true,
+        //    then this operation failed with a "ConstraintError" DOMException.
+        //    Abort this algorithm without taking any further steps.
+        if (index_multi_entry && index_key_is_array && index_is_unique) {
+            for (auto const& subkey : index_key_value->subkeys()) {
+                if (index->has_record_with_key(*subkey))
+                    return WebIDL::ConstraintError::create(realm, "Record already exists in index"_string);
+            }
+        }
+
+        // 5. If index’s multiEntry flag is false, or if index key is not an array key
+        //    then store a record in index containing index key as its key and key as its value.
+        //    The record is stored in index’s list of records such that the list is sorted primarily on the records keys,
+        //    and secondarily on the records values, in ascending order.
+        if (!index_multi_entry || !index_key_is_array) {
+            // FIXME:
+            // Record index_record = {
+            //     .key = index_key_value,
+            //     .value = MUST(HTML::structured_serialize_for_storage(realm.vm(), key)),
+            // };
+            // index->store_a_record(index_record);
```

Comment:
would prefer if these fixmes just stated "store a record whose key is blah and value is blahblah" instead of having commented out code.